dll/win32/imm32/ime.c

```diff
@@ -355,13 +355,184 @@ BOOL APIENTRY Imm32ReleaseIME(HKL hKL)
     return ret;
 }
 
+#define MAX_IMEMENU_BITMAP_BYTES 0x1000
+
+typedef struct tagIMEMENUITEM
+{
+    IMEMENUITEMINFOW Info;
+    BYTE abChecked[MAX_IMEMENU_BITMAP_BYTES];
+    BYTE abUnchecked[MAX_IMEMENU_BITMAP_BYTES];
+    BYTE abItem[MAX_IMEMENU_BITMAP_BYTES];
+} IMEMENUITEM, *PIMEMENUITEM;
+
+typedef struct tagIMEMENU
+{
+    DWORD dwVersion;
+    DWORD dwFlags;
+    DWORD dwType;
+    DWORD dwItemCount;
+    IMEMENUITEMINFOW Parent;
+    IMEMENUITEM Items[ANYSIZE_ARRAY];
+} IMEMENU, *PIMEMENU;
+
+/***********************************************************************
+ *		ImmPutImeMenuItemsIntoMappedFile (IMM32.@)
```

Comment:
Having an actual decent doxygen documentation block that explain what the purpose of all these otherwise undocumented functions that are mysteriously added with no explanations, would be welcome.
(Instead of this stupid Wine-type header that doesn't bring any plus-value.)

Reply:
Comments added.